kufar/books/main.go

```diff
@@ -358,25 +369,31 @@ func ReadBook(d Database) func(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		row := d.db.QueryRow("select id, name, author, date, userid from books where id = $1", idInt)
+		c := NewMap()
```

Comment:
You must init this cache only once because in this case it will be always empty on every read request